benchmarks/urls.py

```diff
@@ -3,41 +3,40 @@
-    path('display-name/vision', user.DisplayName.as_view(domain="vision"), name='display-name-vision'),
-    path('display-name/language', user.DisplayName.as_view(domain="language"), name='display-name-language'),
-    path('profile/vision/', user.Profile.as_view(domain="vision"), name='vision-information'),
-    path('profile/language/', user.Profile.as_view(domain="language"), name='language-information'),
-    path('profile/vision/submit/', user.Upload.as_view(domain="vision"), name='vision-submit'),
-    path('profile/language/submit/', user.Upload.as_view(domain="language"), name='language-submit'),
-    path('profile/vision/resubmit/', user.resubmit, name='vision_resubmit'),
-    path('profile/language/resubmit/', user.resubmit, name='language_resubmit'),
-    path('profile/vision/logout/',  user.Logout.as_view(domain="vision"), name='vision-logout'),
-    path('profile/language/logout/', user.Logout.as_view(domain="language"), name='language-logout'),
-    path('model/vision/<int:id>', model.view, name='model-vision'),
-    path('model/language/<int:id>', model.view, name='model-language'),
+# all currently supported Brain-Score domains:
+supported_domains = ["vision", "language"]
 
+non_domain_urls = [
+        path('', functools.partial(index, domain='vision'), name='index'),
+        path('/', functools.partial(index, domain='vision'), name='index'),
+        # user
+        path('logout/', user.Logout.as_view(domain="vision"), name='logout'),
+        path('upload/', user.Profile.as_view(domain="vision"), name='upload'),
+        path('public-ajax/', user.PublicAjax.as_view(), name='PublicAjax'),
+        path('competition/', competition.view, name='competition'),
+
+        # default profile is vision
+        path('profile/', user.Profile.as_view(domain="vision"), name='default-profile'),
 ]
+
+all_domain_urls = [non_domain_urls]
+
+for domain in supported_domains:
+    domain_urls = [
+        path(f'{domain}/', functools.partial(index, domain=domain), name='index'),
+        path(f'password/{domain}/',  user.Password.as_view(domain=domain), name=f'password-{domain}'),
+        path(f'password-change/{domain}/<str:uidb64>/<str:token>', user.ChangePassword.as_view(domain=domain), name=f'change-password-{domain}'),
+        path(f'signup/{domain}', user.Signup.as_view(domain=domain), name=f'{domain}-signup'),
+        path(f'activate/{domain}/<str:uidb64>/<str:token>', user.Activate.as_view(domain=domain), name=f'activate-{domain}'),
+        path(f'display-name/<str:domain>', user.DisplayName.as_view(domain=domain), name=f'display-name'),
+        path(f'profile/{domain}/', user.Profile.as_view(domain=domain), name=f'{domain}-information'),
+        path(f'profile/{domain}/submit/', user.Upload.as_view(domain=domain), name=f'{domain}-submit'),
+        path(f'profile/<str:domain>/resubmit/', functools.partial(user.resubmit, domain=domain), name=f'resubmit'),
```

Comment:
why `<str:domain>` instead of `{domain}` here?

Reply:
This is a weird case of form action URLs needing to be dynamic, i.e.   

`<form action="{% url 'display-name' domain %}" method="post">`

In this case, HTML does not allow a variable name to passed to a form, so this does not work:

`<form action="{% url 'display-name-{{ domain }}' %}" method="post">`

So this is a workaround, found mainly [here](https://stackoverflow.com/questions/69876942/django-dynamic-url-parameters)